src/main/java/com/amazon/dlic/util/RolesUtil.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.dlic.util;
+
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RolesUtil {
+    private static final Logger log = LogManager.getLogger(RolesUtil.class);
+
+    /*
+     * Split the given rolesObject to one or several roles
+     * We accept String or JSONarray only
+     *
+     * rolesObject should be String if given by roles_key
+     * rolesObject should be String or ArrayList if given by roles_path
+     *
+     * Here are some examples:
+     *
+     *   JWT paylaoad contains:
+     *     "access": {
+     *       "roles": [
+     *         "readall, testrole",
+     *         "admin",
+     *         123,
+     *         "kibanauser"
+     *       ]
+     *     }
+     *  if contains a Json array, it should return elements that is a String.
+     *    roles_path = $["access"]["roles"]
+     *    should return ["readall, testrole", "admin", "kibanauser"]
+     *
+     *  if contains a String, it will split by comma to get roles.
+     *    roles_path = $["access"]["roles"][0]
+     *    should return ["readall", "testrole"]
+     *
+     *  if contains neither Json array nor String, e.g. Json Object, it should throw exception.
+     *    roles_path = $["access"]
+     *    should throw ElasticsearchSecurityException
+     */
+    public static String[] split(Object rolesObject) {
+        if (rolesObject == null) {
+            return new String[0];
+        } else if (rolesObject instanceof ArrayList) {
```

Comment:
Can it be another implementation of the `List`?

Reply:
We use `com.jayway.jsonpath` to parse JSON. If the input is ` io.jsonwebtoken.Claims`, it returns `ArrayList` for JSON array. If the input is `String`, it returns `LinkedList` for JSON array. So yes, it could be another implementation of `List` though we only call this function `split` with `ArrayList` in `HTTPJwtAuthenticator` class. However, in `AbstractHTTPJwtAuthenticator`, which is used by OIDC, this function `split` is called with `LinkedList`. I included the changes in the next PR for OIDC

Reply:
My point is that it is not necessary to rely on `ArrayList` or `LinkedList`, check for `List`.

Reply:
Got it. Yes, it is better to implement with `List`. I will fix it.